src/rgw/rgw_lc.cc

```diff
@@ -582,7 +586,7 @@ static int remove_expired_obj(
       "ERROR: publishing notification failed, with error: " << ret << dendl;
   } else {
     // send request to notification manager
-    (void) notify->publish_commit(dpp, obj->get_obj_size(),
+    (void) notify->publish_commit(dpp, obj_state->size,
 				  ceph::real_clock::now(),
 				  obj->get_attrs()[RGW_ATTR_ETAG].to_str(),
```

Comment:
should we change this line as well to:
```
obj_state->attrset[RGW_ATTR_ETAG].to_str()
```

Reply:
I will add that (thanks for contributing this!)